### PR TITLE
[mesheryctl] Chaining token functions for authentication

### LINKED ISSUE #4805: mesheryctl system config aks: Error getting context: could not read token:: open /Users/me/.meshery/auth.json: no such file or directory
mesheryctl  `v0.6.0-rc-2`

Following this guide: https://docs.meshery.io/installation/platforms/aks

```
13:34 $ mesheryctl system config aks
Missing Meshery config file.
Create default config now [y/n]? y
Default config file created at /Users/me/.meshery/config.yaml
{
  "azure-cli": "2.24.1",
  "azure-cli-core": "2.24.1",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "azure-batch-cli-extensions": "6.0.0",
    "storage-preview": "0.7.2"
  }
}
Configuring Meshery to access AKS...
Please enter the Azure resource group name:
me-rg
Please enter the AKS cluster name:
me-test-001-aks
Merged “me-test-001-aks" as current context in /Users/me/.meshery/kubeconfig.yaml
Error getting context: could not read token:: open /Users/me/.meshery/auth.json: no such file or directory
```

### PR DESCRIPTION
**Pull request by hexxdump**
**Description**
`mesheryctl system config ...` command throws an error message, when incorrect/no token file is provided.
Ideally, the config command should be run only after mesheryctl is authenticated to meshery server using `mesheryctl system login` command.

This PR fixes #4805 

Signed-off-by: Hussaina Begum Nandyala <hexxdump@gmail.com>

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by piyushsingariya** [comment]
@hexxdump we can skip that prereqcheck for now, and think back at that when the refactoring of mesheryctl comes into action.

**Comment by hexxdump** [comment]
> @hexxdump we can skip that prereqcheck for now, and roll back to that when the refactoring of mesheryctl comes into action.

sure @piyushsingariya. We can go ahead with these changes for now.

**Comment by leecalcote** [comment]
Can we make sure that we have centralized onto the same token check mechanism and same user-facing message? Do we have need for more than one token validation check? Here's a sample from `perf`:

```
➜  mesheryctl git:(master) ./mesheryctl perf profile sam
Error: authentication token not found. please supply a valid user token with the --token (or -t) flag. or login with `mesheryctl system login`.
See https://docs.meshery.io/reference/mesheryctl/perf/profile for usage details
```

This message used here is different than that proposed in this PR. What's the plan to getting to a single function for token validation?

**Comment by piyushsingariya** [comment]
> Can we make sure that we have centralized onto the same token check mechanism and same user-facing message? Do we have need for more than one token validation check? Here's a sample from `perf`:
> 
> ```
> ➜  mesheryctl git:(master) ./mesheryctl perf profile sam
> Error: authentication token not found. please supply a valid user token with the --token (or -t) flag. or login with `mesheryctl system login`.
> See https://docs.meshery.io/reference/mesheryctl/perf/profile for usage details
> ```
> 
> This message used here is different than that proposed in this PR. What's the plan to getting to a single function for token validation?

This error isn't from a token function, it's from the part where the request made to meshery-server fails.

**Comment by leecalcote** [comment]
@piyushsingariya thanks for confirming. Will the change in this PR elicit the same error message from the server? Can we make sure that we have centralized onto the same token check mechanism and same user-facing message? Do we have need for more than one token validation check?

**Comment by piyushsingariya** [comment]
> @piyushsingariya thanks for confirming. Will the change in this PR elicit the same error message from the server? Can we make sure that we have centralized onto the same token check mechanism and same user-facing message? Do we have need for more than one token validation check?

@leecalcote and @hexxdump I'm taking over this PR, and implementing an idea that will chain all the token functions and will decrease repetitive code all over the place(I guess).

**Comment by piyushsingariya** [comment]
```
➜  mesheryctl git:(master) ./mesheryctl perf profile sam
Error: authentication token not found. please supply a valid user token with the --token (or -t) flag. or login with `mesheryctl system login`.
See https://docs.meshery.io/reference/mesheryctl/perf/profile for usage details
```

@leecalcote now this error is centralized and will be thrown at all places.

and damn, what have I done to this PR.

@hexxdump This PR is now one step closer to centralizing and chaining mesheryctl functions.

**Comment by piyushsingariya** [comment]
@hexxdump time to review your own PR :joy:

File: mesheryctl/pkg/utils/helpers.go [hunk 1]
```diff
@@ -827,3 +806,15 @@ func SetOverrideValues(ctx *config.Context, mesheryImageVersion string) map[stri
...
+// CheckFileExists checks if the given file exists in system or not
+func CheckFileExists(name string) (bool, error) {
+	_, err := os.Stat(name)
+	if err == nil {
+		return true, nil
+	}
+	if errors.Is(err, os.ErrNotExist) {
+		return false, fmt.Errorf("%s does not exist", name)
+	}
+	return false, errors.Wrap(err, fmt.Sprintf("%s does not exist", name))
```
Comment (hexxdump) [inline mesheryctl/pkg/utils/helpers.go:819]: should this error statement be something generic, as the ErrNotExist case is already handled in line:817
For e.g. Failed to read/fetch the file?

File: mesheryctl/internal/cli/root/app/view_test.go [hunk 2]
```diff
@@ -92,7 +93,7 @@ func TestAppView(t *testing.T) {
 			apiResponse := utils.NewGoldenFile(t, tt.Fixture, fixturesDir).Load()
 
 			// set token
-			tokenPath = tt.Token
+			constants.TokenFlag = tt.Token
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/app/view_test.go:96]: Does it make sense to add this flag under a different package or import the constant package with some other alias? This sounds like we are modifying the constants.

Comment (piyushsingariya) [reply mesheryctl/internal/cli/root/app/view_test.go:96]: So the new logic of creating a http request does two things
1. search Token passed by `--token` that has been assigned to var constants.TokenFlag
2. If found ""; i.e. no token was passed with flag, location is fetched from meshconfig
3. Checks if the file exists
4. Try adding the file to request

returns the http request.

During unit tests, we mock server response but token needs to be attached to due to the requirement of the logic, so we save a sample token but it doesn't exists at the location defined in the meshconfig, so two things could be done here; pass the token location with "--token" flag or set the value of `constants.TokenFlag` manually, and passing the flag does the same things so chose wrote it manually.

File: mesheryctl/pkg/utils/helpers.go (hunk 1 above)
Comment (piyushsingariya) [reply mesheryctl/pkg/utils/helpers.go:819]: Good point, will make the changes tomorrow.

File: mesheryctl/internal/cli/root/app/view_test.go (hunk 2 above)
Comment (hexxdump) [reply mesheryctl/internal/cli/root/app/view_test.go:96]: @piyushsingariya, I understand the reasons behind, manually storing tt.Token value in constants.TokenFlag
The variables/flags defined in constants.go must be used for read only purpose. 
Ideally, any flag/field which can be modified later, should belong to a different package. 
May be the TokenFlag field should move to either internal/cli/root/config package or pkg/utils package.

Comment (piyushsingariya) [reply mesheryctl/internal/cli/root/app/view_test.go]: makes sense, moved to utils package

**Review by hexxdump — COMMENTED** [review]
LGTM

**Review by piyushsingariya — APPROVED** [review]
Let's hope this doesn't break anything. I tried one subcommand under each command to test if the token flag is working correctly.

**Comment by piyushsingariya** [comment]
jk, I have tested the commands myself.

**Comment by hexxdump** [comment]
> jk, I have tested the commands myself.

CITY @piyushsingariya :-)

**Pull request merged**